src/javascripts/services/transactionService.js

```diff
@@ -0,0 +1,43 @@
+import { API_BASE_URL } from "../constants/urls";
+import Transaction from "../models/transactionModel";
+import ApiService from "./apiService";
+
+class TransactionService {
+    constructor(){
+        this.api = new ApiService(API_BASE_URL, '/transaction');
+        this.transactionList;
+        this.initTransactionList();
+    }
+
+    /**
+     * Initializing the Transactions object
+     */
+    initTransactionList = async () => {
+        const data = await this.api.get();
+        this.transactionList = this.parseData(data);
+    }
+    
+    /**
+     * Parsing data from JSON object to list of Transaction object
+     * @param {JSON} data
+     */
+    parseData = (data) => {
+        return data.map((item) => new Transaction(item));
+    }
+
+    /**
+     * Get Transaction objects list
+     * @returns {array}
+     */
+    getTransactions() {
+        return this.transactionList;
+    }
+
+    addTransaction = async(data) => {
+        const transaction = new Transaction(data);
+        console.log(transaction);
+        await this.api.addTransaction(transaction);
```

Comment:
What if this failed?

Reply:
Do you mean that if this failed, an error message will be displayed ?. Sorry because I don't understand you very well